src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -163,24 +165,47 @@ public static double BitIncrement(double x)
             return BitConverter.Int64BitsToDouble(bits);
         }
 
-        public static unsafe double CopySign(double x, double y)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static double CopySign(double x, double y)
         {
-            // This method is required to work for all inputs,
-            // including NaN, so we operate on the raw bits.
+            const long signMask = 1L << 63;
 
-            long xbits = BitConverter.DoubleToInt64Bits(x);
-            long ybits = BitConverter.DoubleToInt64Bits(y);
+            if (Sse.X64.IsSupported)
```

Comment:
Might be able to optimize this with `BlendVariable` on SSE4.1+

Reply:
On it rn :+1: